pulumi/infra/nomad_server_fleet.py

```diff
@@ -0,0 +1,181 @@
+                    "Statement": [
+                        {
+                            "Action": "sts:AssumeRole",
+                            "Effect": "Allow",
+                            "Principal": {
+                                "Service": "ec2.amazonaws.com",
+                            },
+                        }
+                    ],
+                }
+            ),
+            opts=child_opts,
+        )
+
+        self._open_initial_ports(internal_service_ports)
+
+        # InstanceProfile for nomad-server instances
+        instance_profile = aws.iam.InstanceProfile(
+            f"{name}-instance-profile",
+            opts=child_opts,
+            role=self.role.name,
+            name=f"{DEPLOYMENT_NAME}-nomad-server-instance-profile",
+        )
+
+        # CloudWatchAgentServerPolicy allows the Nomad server instances to
+        # run the CloudWatch Agent.
+        policies.attach_policy(
+            role=self.role, policy=policies.CLOUDWATCH_AGENT_SERVER_POLICY
+        )
+        policies.attach_policy(role=self.role, policy=policies.SSM_POLICY)
+
+        policies.attach_policy(role=self.role, policy=EC2_DESCRIBE_INSTANCES_POLICY)
+
+        nomad_server_ami = get_ami_id("grapl-nomad-consul-server")
+
+        nomad_servers = Ec2Cluster(
+            name="nomad-servers",
+            vpc=network,
+            quorum_size=3,
+            quorums=1,
+            ami=nomad_server_ami,
+            instance_type="t2.micro",
+            iam_instance_profile=instance_profile,
+            vpc_security_group_ids=[self.security_group.id],
+            instance_tags={
+                "nomad-server-sec-group-for-deployment": DEPLOYMENT_NAME,
+                "ConsulAgent": "Server",
+            },
+            opts=child_opts,
+        )
+
+    def _open_initial_ports(self, internal_service_ports: Sequence[Ec2Port]) -> None:
+        # allow hosts in the nomad-server security group to communicate
+        # internally on the following ports:
+        # https://www.nomadproject.io/docs/install/production/requirements#ports-used
+        # https://www.consul.io/docs/install/ports
+        for port in (
+            # Nomad ports
+            Ec2Port("tcp", 4646),
+            Ec2Port("tcp", 4647),
+            Ec2Port("tcp", 4648),
+            Ec2Port("udp", 4648),
+            # Consul ports
+            Ec2Port("tcp", 8300),
+            Ec2Port("tcp", 8301),
+            Ec2Port("udp", 8301),
+            Ec2Port("tcp", 8302),
+            Ec2Port("udp", 8302),
+            Ec2Port("tcp", 8500),
+            Ec2Port("udp", 8600),
+            Ec2Port("tcp", 8600),
+            *internal_service_ports,
+        ):
+            port.allow_internally(self.security_group)
+
+        # allow hosts in the nomad-server security group to make outbound
+        # connections to the Internet for these services:
+        #   TCP 443 -- AWS SSM Agent (for handshake)
+        #   TCP 80 -- yum package manager and wget
```

Comment:
Why does it need to run `yum` or `wget`?

Reply:
it doesn't, looks like cobrien copy pasted this